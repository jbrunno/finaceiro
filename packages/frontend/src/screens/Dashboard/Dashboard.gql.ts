import { gql } from '@apollo/client';
import { DashboardAcordosPeriodo } from './components/DashboardAcordosPeriodo/DashboardAcordosPeriodo';
import { DashboardIcm } from './components/DashboardIcm/DashboardIcm';
import { DashboardProducaoGrafico } from './components/DashboardProducaoGrafico/DashboardProducaoGrafico';

import type { IcmData } from './components/DashboardIcm/DashboardIcm.gql';
import type { ProducaoAcordosComparativoData } from './components/DashboardProducaoGrafico/DashboardProducaoGrafico.gql';
import type { ProducaoAcordosSintetizadoData } from './components/DashboardAcordosPeriodo/DashboardAcordosPeriodo.gql';

export const ICM_QUERY = gql`
  query Icm {
    icm {
      ...DASHBOARD_ICM_FRAGMENT
    }
  }

  ${DashboardIcm.fragments.icm}
`;

export const DASHBOARD_PRODUCAO_GRAFICO_QUERY = gql`
  query ProducaoAcordosComparativo {
    producaoAcordosComparativo {
      ...DASHBOARD_PRODUCAO_GRAFICO_FRAGMENT
    }
  }

  ${DashboardProducaoGrafico.fragments.producaoAcordosComparativo}
`;

export const DASHBOARD_ACORDOS_PERIODO_QUERY = gql`
  query ProducaoAcordosSintetizado($date: Date!) {
    producaoAcordosSintetizado(date: $date) {
      ...DASHBOARD_ACORDOS_PERIODO_FRAGMENT
    }
  }
  ${DashboardAcordosPeriodo.fragments.producaoAcordosSintetizado}
`;

export type DashboardAcordosPeriodosVariables = {
  date: string;
};

export type IcmQueryResult = {
  icm: IcmData;
};

export type AcordosComparativosQueryResult = {
  producaoAcordosComparativo: ProducaoAcordosComparativoData;
};

export type AcordosSinterizadosQueryResult = {
  producaoAcordosSintetizado: ProducaoAcordosSintetizadoData;
};
