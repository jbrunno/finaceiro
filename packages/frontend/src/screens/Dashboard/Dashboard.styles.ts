import { styled } from '@frontend/styles';

import { DashboardAcordosPeriodo } from './components/DashboardAcordosPeriodo/DashboardAcordosPeriodo';
import { DashboardIcm } from './components/DashboardIcm/DashboardIcm';
import { DashboardProducaoGrafico } from './components/DashboardProducaoGrafico/DashboardProducaoGrafico';

export const DashboardIcmStyled = styled(DashboardIcm)`
  grid-column: span 4;
`;

export const DashboardAcordosPeriodoStyled = styled(DashboardAcordosPeriodo)`
  grid-column: span 8;
`;

export const DashboardProducaoGraficoStyled = styled(DashboardProducaoGrafico)`
  grid-column: span 12;
`;

export const Widgets = styled('div')`
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  grid-template-rows: auto 1fr;
  gap: ${({ theme }) => theme.fn.spacing(16)};
  height: 100%;
  min-height: 0;
`;
