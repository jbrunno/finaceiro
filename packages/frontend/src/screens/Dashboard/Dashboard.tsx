import { PageHeader, RenderComponents } from '@frontend/components';
import { Content } from '@frontend/styles';
import {
  DashboardAcordosPeriodoStyled,
  DashboardProducaoGraficoStyled,
  DashboardIcmStyled,
  Widgets,
} from './Dashboard.styles';
import { useDashboard } from './hooks/useDashboard';

export function Dashboard() {
  const { data, setDataPeriodo } = useDashboard();
  return (
    <Content>
      <PageHeader title="Dashboard" />
      <Widgets>
        <RenderComponents screen="dashboard">
          <DashboardIcmStyled perm="DASHBOARD_ICM" icm={data?.icm} />
          <DashboardAcordosPeriodoStyled
            perm="DASHBOARD_ACORDOS_PERIODO"
            onChangePeriod={setDataPeriodo}
            producaoAcordosSintetizado={data?.producaoAcordosSintetizado}
          />
          <DashboardProducaoGraficoStyled
            perm="DASHBOARD_PRODUCAO_GRAFICO"
            producaoAcordosComparativo={data?.producaoAcordosComparativo}
          />
        </RenderComponents>
      </Widgets>
    </Content>
  );
}
