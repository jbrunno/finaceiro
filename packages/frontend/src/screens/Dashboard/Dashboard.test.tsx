import { render } from '@test-utils/component-testing';
import { Dashboard } from './Dashboard';

test('should render Dashboard', () => {
  const { container } = render(<Dashboard />);
  expect(container).toMatchSnapshot();
});
