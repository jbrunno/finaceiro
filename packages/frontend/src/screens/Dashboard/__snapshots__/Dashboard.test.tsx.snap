// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Dashboard 1`] = `
.emotion-0 {
  display: grid;
  grid-template-rows: auto;
  grid-auto-flow: row;
  grid-auto-rows: 1fr;
  row-gap: 1rem;
  margin-bottom: 1rem;
  height: 100%;
}

.emotion-2 {
  display: grid;
  margin-top: 1rem;
  row-gap: 0.25rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-auto-rows: min-content;
  grid-auto-flow: row;
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-7 {
  margin: 0;
  font-weight: 700;
  font-size: 1.5rem;
  line-height: 1.8125rem;
  letter-spacing: 0;
  font-family: Lato,sans-serif;
  color: #3B3B41DE;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-8 {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  grid-template-rows: auto 1fr;
  gap: 1rem;
  height: 100%;
  min-height: 0;
}

<div>
  <section
    class="emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <div
        class="MuiGrid-root MuiGrid-container emotion-4"
      >
        <div
          class="MuiGrid-root MuiGrid-item emotion-5"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 emotion-6 emotion-7"
          >
            Dashboard
          </h3>
        </div>
      </div>
    </div>
    <div
      class="emotion-8 emotion-9"
    />
  </section>
</div>
`;
